### PR TITLE
4 read in the family resources survey

### LINKED ISSUE #4: Read in the family resources survey
Translate the FRS from Stata to csv, do any relevant cleaning and preprocessing and upload back to S3. Create the pipelines to read the data into projects.

### PR DESCRIPTION
**Pull request by VolcanoBlue13**
---

# Description

Functions to perform basic cleaning on the FRS 2022/2023 data including reading in from the Stata file format and renaming columns and converting these to csvs.

Fixes #4-read-in-the-family-resources-survey

# Instructions for Reviewer

If you could just see whether you can read in the FRS data, that would be fantastic, thanks @j-gillam! Most of the columns should have more clear names, but some still won't have (haven't worked out what they are yet).

# Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [x] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [x] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [x] I have explained this PR above
- [x] I have requested a code review

### CODE REVIEW

**Review by j-gillam — COMMENTED** [review]
Hello! Think this looks great! Some small comments that you can decide if you want to do or not :)

File: afs_mission_goal/getters/uk_data_service/raw/family_resources_survey.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
+import pandas as pd
```
[the remark this answers is not in the document]
Comment (VolcanoBlue13) [reply afs_mission_goal/getters/uk_data_service/raw/family_resources_survey.py:1]: > Excellent idea! Have done this

File: afs_mission_goal/getters/uk_data_service/processed/family_resources_survey.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+"""
+To read in the Family Resources Survey datasets from the UK Data Service, you have the option of two functions. One reads in every dataset into a dictionary where the key is the dataset name and the value is the pd.DataFrame. The second function allows you to read in individual datasets, with an argument to say which dataset you want to read in.
+"""
+
+import pandas as pd
+from nesta_ds_utils.loading_saving.S3 import download_obj
+
+
+BUCKET = "afs-uk-data-service"
+frs_datasets = [
```
[the remark this answers is not in the document]
Comment (VolcanoBlue13) [reply afs_mission_goal/getters/uk_data_service/processed/family_resources_survey.py:10]: That's a good idea, have created a config file and added into init!

File: afs_mission_goal/pipeline/clean_family_resources_survey.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        "owner",
+        "pension_provider",
+        "pension",
+        "rent_cont",
+        "renter",
+        "tables",
+    ]
+
+    for dataset in frs_datasets:
+        frs_data = globals()[f"get_{dataset}_data"]()
```
[the remark this answers is not in the document]
Comment (VolcanoBlue13) [reply afs_mission_goal/pipeline/clean_family_resources_survey.py:58]: I think it was a way of calling a function which had similar names in a loop, but I've solved this by removing the individual functions and creating just one as you suggested!

File: afs_mission_goal/pipeline/clean_family_resources_survey.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+frs_columns = get_family_resources_survey_dict()
+
+
+def clean_family_resources_survey(frs_data: pd.DataFrame) -> pd.DataFrame:
+    new_columns = []
+    count = 0
+
+    for col in frs_data.columns:
+        try:
+            new_columns.append(frs_dict[col])
```
[the remark this answers is not in the document]
Comment (VolcanoBlue13) [reply afs_mission_goal/pipeline/clean_family_resources_survey.py]: Good catch, have changed!

File: afs_mission_goal/pipeline/clean_family_resources_survey.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import pandas as pd
+from afs_mission_goal.getters.uk_data_service.raw.family_resources_survey import *
+from afs_mission_goal.getters.uk_data_service.misc.get_family_resources_survey_dict import (
+    get_family_resources_survey_dict,
+)
+from afs_mission_goal.utils.preprocessing import preprocess_strings
+
+from nesta_ds_utils.loading_saving import S3
+
+frs_columns = get_family_resources_survey_dict()
```
[the remark this answers is not in the document]
Comment (VolcanoBlue13) [reply afs_mission_goal/pipeline/clean_family_resources_survey.py:10]: Good idea, have moved!

**Pull request merged**